hyperframe/frame.py

```diff
@@ -381,20 +391,21 @@ def __init__(self, stream_id=0, settings=None, **kwargs):
         self.settings = settings or {}
 
     def serialize_body(self):
-        settings = [struct.pack("!HL", setting & 0xFF, value)
-                    for setting, value in self.settings.items()]
-        return b''.join(settings)
+        return b''.join([_STRUCT_HL.pack(setting & 0xFF, value)
+                         for setting, value in self.settings.items()])
```

Comment:
Well that's astonishingly disappointing. A quick check of the CPython codebase reveals that you're correct, .join() creates an intermediate list anyway. Presumably this cost comes from the fact that the intermediate generator needs to be stored and generally handled, whereas the list comprehension optimises that.

Ugh, that's really annoying. Good spot!